core/trino-main/src/main/java/io/trino/execution/buffer/SpoolingExchangeOutputBuffer.java

```diff
@@ -139,7 +142,11 @@ public void destroy(OutputBuffers.OutputBufferId bufferId)
     @Override
     public ListenableFuture<Void> isFull()
     {
-        return toListenableFuture(exchangeSink.isBlocked());
+        ExchangeSink sink = exchangeSink;
+        if (sink != null) {
```

Comment:
can `isFull` be called after close? Should it be `requireNotNull`?

Reply:
It can be called after `abort`. The `isFull` method is called from the main `Driver` processing loop, while abort can be called by the `TaskResource` to terminate a task early.